enum.rb

```diff
@@ -0,0 +1,521 @@
+# frozen_string_literal: true
+
+#:nodoc:
+# rubocop: disable Metrics/ModuleLength
+module Enumerable
+  # *********************************
+  # Beginning of my_each method.
+  # *********************************
+  # rubocop: disable Metrics/AbcSize,Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+  def my_each(nums = 0)
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`each`](https://ruby-doc.org/core-2.6.4/Array.html#method-i-each). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] my_each returns an Enumerator if no block is given
```ruby
array = Array.new(100) { rand(0...9) } }
array.my_each) == Enumerator #1 true
```

Reply:
Tests tun successfully